### PR TITLE
Button code

### PR DESCRIPTION
**Pull request by arniscg**

### CODE REVIEW

File: controller/src/main.c [hunk 1]
```diff
@@ -9,52 +9,214 @@
...
+
+	time = last_time;
+
+    send_button_value(0);
+	
 }
 
-void main(void)
+
+void button_pressed1(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
```
Comment (arniscg) [inline controller/src/main.c:66]: I don't think we need a button press handler function for every button. This leads to a lot of code duplication.

We should be able to give the same handler function to `gpio_init_callback()` and then in that handler we can know which button was pressed based on function parameters. For example from device struct pointer `*dev`.

File: controller/src/main.c [hunk 2]
```diff
@@ -9,52 +9,214 @@
...
+	if (veml7700 == NULL) {
+		printf("No device \"%s\" found; did initialization fail?\n", "VEML7700");
+		return;
 	}
+
+	while(1)
+	{
+	 	  
+			
+			k_sleep(K_SECONDS(3));
```
Comment (arniscg) [inline controller/src/main.c:178]: You have removed sensor value reading and sending logic.

File: controller/src/main.c [hunk 3]
```diff
@@ -9,52 +9,214 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (arniscg) [inline controller/src/main.c:222]: It's not very pretty to leave 20+ empty lines at the end of the file :)

File: controller/src/main.c (hunk 1 above)
Comment (Sukirti-Singh) [reply controller/src/main.c:66]: I have tried that but I was comparing dev it was not detecting different button. device_get_binding(SW0) == dev comparing.
I

Comment (arniscg) [reply controller/src/main.c:66]: Yeah, you are right, seems like dev simply points to GPIO_0.

But I managed to do it by using `pins`. Looks like this:
```
void button_handler(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
{
	time = k_uptime_get_32();
	if (time < last_time + BUTTON_DEBOUNCE_DELAY_MS) 
	{
		last_time = time;
		return;
	}

	time = last_time;

	switch (pins)
	{
		case BIT(SW0_PIN):
			printk("Button 0 pressed\n");
			send_button_value(0);
			break;
		case BIT(SW1_PIN):
			printk("Button 1 pressed\n");
			send_button_value(1);
			break;
		case BIT(SW2_PIN):
			printk("Button 2 pressed\n");
			send_button_value(2);
			break;
		case BIT(SW3_PIN):
			printk("Button 3 pressed\n");
			send_button_value(3);
			break;
		default:
			break;
	}
}
```

File: controller/src/main.c (hunk 2 above)
Comment (Sukirti-Singh) [reply controller/src/main.c:178]: when i have started working sensor code wasn't merged. Do i need to add manually everything?

Comment (arniscg) [reply controller/src/main.c:178]: Yes. Look at the "Files changed" section in this pull request and see what exactly you have removed and then put the missing parts back in. Then push your changes again. This pull request will be updated automatically.

File: controller/src/main.c (hunk 1 above)
Comment (Sukirti-Singh) [reply controller/src/main.c:66]: Now i have made only one handler

**Pull request merged**